modules/local/rna_downsample.nf

```diff
@@ -0,0 +1,56 @@
+process RNA_DOWNSAMPLE {
+    tag "$meta.id"
+    label 'process_low'
+
+    conda "bioconda::samtools=1.16.1"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/samtools:1.16.1--h6899075_1' :
+        'quay.io/biocontainers/samtools:1.16.1--h6899075_1' }"
+
+    input:
+    tuple val(meta), path(bam)
```

Comment:
```suggestion
    tuple val(meta), path(bam), path(bai)
```

Reply:
idxstats is much quicker if you supply the index for the bam as well (which you already have).
http://www.htslib.org/doc/samtools-idxstats.html